dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/PublishingEndPoint.java

```diff
@@ -134,16 +136,15 @@ public StringBuilder toURL() {
         }
         return sb;
     }
-    
+
     /**
-     * Get the endpoint publisher class
-     * @return the endpoint publisher class
+     * TODO
+     * @return
      */
-    public Class getPublisher() {
-        if ( AWSS3Publisher.PROTOCOL_AWS_S3.equals(this.getProtocol()) ) {
-            return AWSS3Publisher.class;
-        } else {
-        	return PushPublisher.class;
-        }
-    }
+    public abstract Class getPublisher();
+
+    /**
+     * TODO
+     */
+    public abstract void validatePublishingEndPoint() throws DotDataException, LanguageException;
```

Comment:
after discussing with @jdotcms and @acurionedotcms, we think it is better to have one new "PublishingEndPointValidationException" that allows to add a list of un-i18n error messages (in a similar fashion to the validateContentlet), so this method will only that new (checked) exception

Reply:
users of the validate need to catch new the exception, get the list of error messages and get the i18n version of them using the user's info